### PR TITLE
Merging functions and reorganized structure

### PR DESCRIPTION
**Pull request by fsoubelet**
This is a **draft** for #56 as well as a small reorganization of the internal structure to be cleaner and closer to `pandas`. It is not breaking as functions are exported to top level.

A `concat` function is also added and exported as `tfs.concat` and the following functions are overwritten from the `pandas.DataFrame`: `append`, `join` and `merge`.

Data manipulation is fully handled by the corresponding `pandas` functions.

## TO BE DONE
- [x] the implementation and choices for headers manipulation is still to be determined @mihofer @JoschD 
- [x] tests are yet to be added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,61 +1,42 @@
...
 
+ The package is imported as `tfs`, and exports top-level functions for reading and writing:
 ```python
 import tfs
 
-data_frame = tfs.read('path_to_input.tfs', index="index_column")
-tfs.write('path_to_output.tfs', data_frame, save_index="index_column")
+data_frame = tfs.read("path_to_input.tfs", index="index_column")
+# do things with the data
+tfs.write("path_to_output.tfs", data_frame, save_index="index_column")
```
Comment (mihofer) [inline README.md:29]: maybe add here how to access headers since this is the distinguishing feature to regular `pd.DataFrame`

**Comment by JoschD** [comment]
I would: 
- [ ] increase major version number, as it breaks it for everyone that used `tfs.handler` etc. in the past
- [ ] update doc with changes from https://github.com/pylhc/PyLHC/pull/76/files

About the headers:
I thought about what I think would make the most sense and then looked what you did. And it was exactly the same. So for me this looks fine.

**Comment by fsoubelet** [comment]
I think this is ready for review now.

- I have done what I could think of in terms of tests, and I will gladly accept any ideas for additional ones on the new functionality!
- I still have not found how to get `sphinx / napoleon` to nicely display the keyword arguments section, so the documentation build looks a bit weird :( Fixes welcome!

Review away.

**Review by mihofer — APPROVED** [review]
Looks good to me. One additional thing/test we could think of is to include a test in the `writer` part where we use either a (latest) mad-x bin or cpymad to load the written `tfs` and see if that still works.

**Comment by fsoubelet** [comment]
As of dc4fd58, the loading of a file written by `tfs-pandas` with `eol` at `eof` into `MAD-X` is complete. 

The presence of the table for appropriate loading is tested, and the tests make use of `pandas.testing.assert_series_equal` to allow for a small relative tolerance when asserting, which `numpy.array_equal` does not allow to test the content of the loaded table is the proper one.

File: tfs/__init__.py [hunk 2]
```diff
@@ -15,4 +18,4 @@
 read = read_tfs
 write = write_tfs
 
-__all__ = [read, write, TfsDataFrame, __version__]
+__all__ = [concat, read, write, TfsDataFrame, __version__]
```
Comment (mihofer) [inline tfs/__init__.py:21]: why only concat? didn't you also add append and merge?

Comment (fsoubelet) [reply tfs/__init__.py:21]: These are methods of `TfsDataFrame`

**Review by mihofer — APPROVED** [review]

**Pull request merged**